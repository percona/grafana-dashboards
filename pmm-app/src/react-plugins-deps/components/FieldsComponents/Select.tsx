import React from 'react';
// import { useField } from 'react-final-form-hooks';
import { Select } from 'antd';
import './Select.scss';

interface SelectFieldInterface {
  required?: boolean;
  name?: string;
  'data-cy'?: string;
  label?: string;
  prefix?: string;
  placeholder?: string;
  form?: any;
  style?: any;
  text?: string;
  options?: any;
  defaultValue?: any;
}
export const SelectField = ({ name, placeholder, required, form, options, defaultValue }: SelectFieldInterface) => {
  // @ts-ignore
  // const field = useField(name, form);
  //   {...field.input}
  return (
    <span className={'fields__select-field'}>
      <Select defaultValue={defaultValue} style={{ width: '40%' }}>
        {options.map(option => {
          // @ts-ignore
          return (
            <Option value={option.value}>
              <div class={'select-item'}>{option.label}</div>
            </Option>
          );
        })}
      </Select>
    </span>
  );
};
