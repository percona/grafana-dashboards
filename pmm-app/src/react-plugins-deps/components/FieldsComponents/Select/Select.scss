#antd {
  .fields__select-field .ant-slider-mark-text {
    color: white;
  }

  .fields__select-field .ant-select-selection__rendered,
  .fields__select-field .ant-select-selection {
    background-color: rgb(20, 20, 20);
    border-radius: 0;
  }

  .fields__select-field .ant-select-selection {
    border: 1px solid rgb(40, 40, 40);
  }

  .fields__select-field .ant-select-focused .ant-select-selection__rendered,
  .fields__select-field .ant-select-focused .ant-select-selection {
    background-color: rgb(20, 20, 20);
  }
  .fields__select-field .ant-select-selection-selected-value {
    color: white;
  }

  .ant-select-dropdown,
  li.ant-select-dropdown-menu-item {
    background: rgb(33, 35, 38) !important;
    color: white !important;
    border-radius: 0;
  }

  .fields__select-field .ant-select-dropdown-menu-item-active .select-item {
    background-color: #cccccc;
  }

  .fields__select-field .ant-select-arrow {
    color: white;
  }
  .ant-select-selection--single {
    height: 40px;
  }
}
