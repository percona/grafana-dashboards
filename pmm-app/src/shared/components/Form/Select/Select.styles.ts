import { css } from 'emotion';

export const styles = {
  select: css`
  width: 100%
  height: 40px
  `,
};
