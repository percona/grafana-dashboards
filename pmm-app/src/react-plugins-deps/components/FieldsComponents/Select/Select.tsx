import React from 'react';
import { useField } from 'react-final-form-hooks';
import { Select } from 'antd';
import './Select.scss';

const { Option } = Select;
interface SelectFieldInterface {
  required?: boolean;
  name: string;
  'data-cy'?: string;
  label?: string;
  prefix?: string;
  placeholder?: string;
  form?: any;
  style?: any;
  text?: string;
  options?: any;
  defaultValue?: any;
}
export const SelectField = ({ name, placeholder, required, form, options, style, defaultValue }: SelectFieldInterface) => {
  const { input } = useField(name, form);
  console.log(input);
  return (
    <span className={'fields__select-field'} style={style || {}}>
      <Select defaultValue={defaultValue} style={{ width: '100%' }}>
        {options &&
          options.map(option => {
            // @ts-ignore
            return (
              <Option value={option.value}>
                <div className={'select-item'}>{option.label}</div>
              </Option>
            );
          })}
      </Select>
    </span>
  );
};
//
// import React from 'react';
// import { useField } from 'react-final-form-hooks';
//
// export const SelectField = ({ name, placeholder, required, form, ...rest }) => {
//   // @ts-ignore
//   const field = useField(name, form);
//
//   return <input type={'password'} {...field.input} placeholder={placeholder} className="input-field input-field--dark" />;
// };
