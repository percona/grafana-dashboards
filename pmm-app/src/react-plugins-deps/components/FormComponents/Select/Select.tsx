import React from 'react';
import { useField } from 'react-final-form';
import { Select } from 'antd';
import './Select.scss';

const { Option } = Select;
interface SelectFieldInterface {
  required?: boolean;
  name: string;
  'data-cy'?: string;
  label?: string;
  prefix?: string;
  placeholder?: string;
  style?: any;
  text?: string;
  options?: any;
  defaultValue?: any;
}
export const SelectField = ({
  name,
  placeholder,
  required,
  options,
  style,
  defaultValue,
}: SelectFieldInterface) => {
  const { input } = useField(name);
  return (
    <span className="fields__select-field" style={style || {}}>
      <Select {...input} defaultValue={defaultValue} style={{ width: '100%', height: '40px' }}>
        {options &&
          options.map(option => {
            // @ts-ignore
            return (
              <Option value={option.value} key={option.label}>
                <div className="select-item">{option.label}</div>
              </Option>
            );
          })}
      </Select>
    </span>
  );
};
