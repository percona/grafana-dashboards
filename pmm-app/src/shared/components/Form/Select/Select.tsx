import React from 'react';
import { useField } from 'react-final-form';
import { Select } from 'antd';
import './Select.scss';
import { styles } from './Select.styles';

const { Option } = Select;

interface SelectFieldInterface {
  required?: boolean;
  name: string;
  dataQa?: string;
  label?: string;
  prefix?: string;
  placeholder?: string;
  style?: any;
  text?: string;
  options?: any;
  defaultValue?: any;
}
export const SelectField = ({
  name, options, style, defaultValue, dataQa,
}: SelectFieldInterface) => {
  const { input } = useField(name);

  return (
    <span className="fields__select-field" style={style || {}}>
      {/* eslint-disable-next-line max-len */}
      <Select {...input} defaultValue={defaultValue} className={styles.select} data-qa={dataQa}>
        {options
          && options.map((option) => (
            <Option value={option.value} key={option.label}>
              <div className="select-item">{option.label}</div>
            </Option>
          ))}
      </Select>
    </span>
  );
};
