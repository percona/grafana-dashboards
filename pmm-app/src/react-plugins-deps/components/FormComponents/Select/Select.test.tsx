// Just a stub test
import { SelectField } from './Select';
import React from 'react';
import renderer from 'react-test-renderer';
import { renderHook } from '@testing-library/react-hooks';
import { Form } from 'react-final-form';

describe('Select field test', () => {
  it('Select renders correct without props', () => {
    renderHook(() => {
      const component = renderer.create(
        <Form onSubmit={() => null} render={() => <SelectField defaultValue="tester" name="test_field" />} />
      );
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
