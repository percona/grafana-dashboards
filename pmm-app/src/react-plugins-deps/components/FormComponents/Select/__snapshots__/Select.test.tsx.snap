// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select field test Select renders correct without props 1`] = `null`;
